packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -86,8 +97,26 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
             throw new PaymentMethodInvalidError();
         }
 
-        await this.paymentIntegrationService.submitOrder(order, options);
-        await this.paypalCommerceIntegrationService.submitPayment(payment.methodId, this.orderId);
+        try {
+            await this.paymentIntegrationService.submitOrder(order, options);
+            await this.paypalCommerceIntegrationService.submitPayment(
+                payment.methodId,
+                this.orderId,
+            );
+        } catch (error: unknown) {
+            if (this.isProviderError(error) && shouldHandleInstrumentDeclinedError && this.methodId) {
```

Comment:
why don't you use `payment.methodId` instead of `this.methodId`?

Reply:
fixed